Can we cache the lyrics in the extension so apps can be run offline?

I notice if I run the quarkus app offline, I get the following failures:

```
Caused by: java.util.concurrent.ExecutionException: java.net.UnknownHostException: api.lyrics.ovh
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at org.example.bon.jova.quarkus.extension.deployment.lyrics.RemoteLyricsReader.lambda$readRemoteLyrics$2(RemoteLyricsReader.java:149)
        ... 24 more
Caused by: java.net.UnknownHostException: api.lyrics.ovh
        at java.base/java.net.InetAddress$CachedLookup.get(InetAddress.java:988)
```

If we baked the lyrics into the extension, the app could run without needing network access. That might make demos more robust.

Obviously, this is only a nice-to-have, and not critical.

--- a/bon-jova-quarkus-extension/deployment/pom.xml
+++ b/bon-jova-quarkus-extension/deployment/pom.xml
@@ -26,10 +26,6 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>io.rest-assured</groupId>
-            <artifactId>rest-assured</artifactId>
-        </dependency>
-        <dependency>
             <groupId>org.example</groupId>
             <artifactId>bon-jova-rockstar-implementation</artifactId>
             <version>1.0-SNAPSHOT</version>
@@ -45,27 +41,10 @@
             <artifactId>quarkus-resteasy-reactive-deployment</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.github.jagrosh</groupId>
-            <artifactId>JLyrics</artifactId>
-            <version>master-SNAPSHOT</version>
-        </dependency>
-        <dependency>
-            <groupId>com.typesafe</groupId>
-            <artifactId>config</artifactId>
-            <version>1.3.2</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.json</groupId>
-            <artifactId>json</artifactId>
-            <version>20180813</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.jsoup</groupId>
-            <artifactId>jsoup</artifactId>
-            <version>1.15.3</version>
-            <type>jar</type>
+            <groupId>org.example</groupId>
+            <artifactId>bon-jova-quarkus-extension-lyrics</artifactId>
+            <version>1.0-SNAPSHOT</version>
+            <scope>compile</scope>
         </dependency>
     </dependencies>
     <build>
@@ -93,27 +72,4 @@
             </plugin>
         </plugins>
     </build>
-    <repositories>
-        <repository>
-            <id>central</id>
-            <name>Central Repository</name>
-            <url>https://repo.maven.apache.org/maven2</url>
-            <layout>default</layout>
-            <snapshots>
-                <enabled>false</enabled>
-            </snapshots>
-        </repository>
-        <repository>
-            <id>jitpack.io</id>
-            <url>https://www.jitpack.io</url>
-            <snapshots>
-                <enabled>true</enabled>
-                <updatePolicy>daily</updatePolicy>
-            </snapshots>
-            <releases>
-                <enabled>true</enabled>
-                <updatePolicy>never</updatePolicy>
-            </releases>
-        </repository>
-    </repositories>
 </project>

--- a/bon-jova-quarkus-extension/deployment/src/main/java/org/example/bon/jova/quarkus/extension/deployment/BonJovaQuarkusExtensionProcessor.java
+++ b/bon-jova-quarkus-extension/deployment/src/main/java/org/example/bon/jova/quarkus/extension/deployment/BonJovaQuarkusExtensionProcessor.java
@@ -13,8 +13,8 @@ import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
 import io.quarkus.devui.spi.page.CardPageBuildItem;
 import io.quarkus.devui.spi.page.Page;
 import org.example.RockFileCompiler;
-import org.example.bon.jova.quarkus.extension.deployment.lyrics.LyricsReader;
 import org.example.bon.jova.quarkus.extension.deployment.rockscore.RockScoreCalculator;
+import org.example.bon.jova.quarkus.extension.lyrics.LyricsReader;
 import org.example.bon.jova.quarkus.extension.runtime.RockstarResource;
 
 import java.io.File;

--- a/bon-jova-quarkus-extension/deployment/src/main/java/org/example/bon/jova/quarkus/extension/deployment/lyrics/LyricsRatingCalculator.java
+++ b/bon-jova-quarkus-extension/deployment/src/main/java/org/example/bon/jova/quarkus/extension/deployment/lyrics/LyricsRatingCalculator.java
@@ -1,6 +1,7 @@
 package org.example.bon.jova.quarkus.extension.deployment.lyrics;
 
-import org.example.bon.jova.quarkus.extension.deployment.wordcounter.WordCounter;
+import org.example.bon.jova.quarkus.extension.lyrics.LyricsSanitizer;
+import org.example.bon.jova.quarkus.extension.lyrics.WordCounter;
 
 import java.util.Arrays;
 import java.util.List;

--- a/bon-jova-quarkus-extension/deployment/src/main/java/org/example/bon/jova/quarkus/extension/deployment/rockscore/RockScoreCalculator.java
+++ b/bon-jova-quarkus-extension/deployment/src/main/java/org/example/bon/jova/quarkus/extension/deployment/rockscore/RockScoreCalculator.java
@@ -1,8 +1,7 @@
 package org.example.bon.jova.quarkus.extension.deployment.rockscore;
 
 import org.example.bon.jova.quarkus.extension.deployment.lyrics.LyricsRatingCalculator;
-import org.example.bon.jova.quarkus.extension.deployment.lyrics.Song;
-import org.example.bon.jova.quarkus.extension.deployment.wordcounter.WordCounter;
+import org.example.bon.jova.quarkus.extension.lyrics.WordCounter;
 
 import java.util.List;
 

--- a/bon-jova-quarkus-extension/deployment/src/test/java/org/example/bon/jova/quarkus/extension/deployment/lyrics/LyricsRatingCalculatorTest.java
+++ b/bon-jova-quarkus-extension/deployment/src/test/java/org/example/bon/jova/quarkus/extension/deployment/lyrics/LyricsRatingCalculatorTest.java
@@ -1,6 +1,7 @@
 package org.example.bon.jova.quarkus.extension.deployment.lyrics;
 
-import org.example.bon.jova.quarkus.extension.deployment.wordcounter.WordCounter;
+import org.example.bon.jova.quarkus.extension.lyrics.LyricsReader;
+import org.example.bon.jova.quarkus.extension.lyrics.WordCounter;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 

--- a/bon-jova-quarkus-extension/deployment/src/test/java/org/example/bon/jova/quarkus/extension/deployment/rockscore/RockScoreCalculatorTest.java
+++ b/bon-jova-quarkus-extension/deployment/src/test/java/org/example/bon/jova/quarkus/extension/deployment/rockscore/RockScoreCalculatorTest.java
@@ -1,6 +1,6 @@
 package org.example.bon.jova.quarkus.extension.deployment.rockscore;
 
-import org.example.bon.jova.quarkus.extension.deployment.lyrics.LyricsReader;
+import org.example.bon.jova.quarkus.extension.lyrics.LyricsReader;
 import org.junit.jupiter.api.Test;
 
 import java.io.IOException;

--- a/bon-jova-quarkus-extension/lyrics/pom.xml
+++ b/bon-jova-quarkus-extension/lyrics/pom.xml
@@ -1,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.example</groupId>
+        <artifactId>bon-jova-quarkus-extension-parent</artifactId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+    <name>Bon Jova Quarkus Extension - Lyrics</name>
+    <artifactId>bon-jova-quarkus-extension-lyrics</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.github.jagrosh</groupId>
+            <artifactId>JLyrics</artifactId>
+            <version>master-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>com.typesafe</groupId>
+            <artifactId>config</artifactId>
+            <version>1.3.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <version>20180813</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jsoup</groupId>
+            <artifactId>jsoup</artifactId>
+            <version>1.15.3</version>
+            <type>jar</type>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5-internal</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <repositories>
+        <repository>
+            <id>jitpack.io</id>
+            <url>https://www.jitpack.io</url>
+            <snapshots>
+                <enabled>true</enabled>
+                <updatePolicy>daily</updatePolicy>
+            </snapshots>
+            <releases>
+                <enabled>true</enabled>
+                <updatePolicy>never</updatePolicy>
+            </releases>
+        </repository>
+    </repositories>
+</project>

--- a/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsFileUtil.java
+++ b/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsFileUtil.java
@@ -1,4 +1,4 @@
-package org.example.bon.jova.quarkus.extension.deployment.lyrics;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
 import java.io.IOException;
 import java.nio.file.Files;

--- a/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsReader.java
+++ b/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsReader.java
@@ -1,4 +1,4 @@
-package org.example.bon.jova.quarkus.extension.deployment.lyrics;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
 import java.util.List;
 import java.util.Optional;

--- a/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsSanitizer.java
+++ b/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsSanitizer.java
@@ -1,4 +1,4 @@
-package org.example.bon.jova.quarkus.extension.deployment.lyrics;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
 import java.util.List;
 import java.util.function.Predicate;

--- a/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/RemoteLyricsReader.java
+++ b/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/RemoteLyricsReader.java
@@ -1,7 +1,9 @@
-package org.example.bon.jova.quarkus.extension.deployment.lyrics;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
 import com.jagrosh.jlyrics.Lyrics;
 import com.jagrosh.jlyrics.LyricsClient;
+import io.restassured.RestAssured;
+import io.restassured.path.json.JsonPath;
 
 import java.util.HashMap;
 import java.util.List;
@@ -207,7 +209,7 @@ public class RemoteLyricsReader {
             var url = "https://api.lyrics.ovh/v1/{artist}/{title}";
             var ignorePattern = "Paroles de la chanson(.+) par (.+)\\r\\n";
 
-            String response = with(get(url, song.artist(), song.title()).asString()).get("lyrics");
+            String response = JsonPath.with(RestAssured.get(url, song.artist(), song.title()).asString()).get("lyrics");
 
             if (response == null) {
                 return Optional.empty();

--- a/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/Song.java
+++ b/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/Song.java
@@ -1,4 +1,4 @@
-package org.example.bon.jova.quarkus.extension.deployment.lyrics;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
 import java.util.Objects;
 import java.util.Optional;

--- a/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/WordCounter.java
+++ b/bon-jova-quarkus-extension/lyrics/src/main/java/org/example/bon/jova/quarkus/extension/lyrics/WordCounter.java
@@ -1,6 +1,4 @@
-package org.example.bon.jova.quarkus.extension.deployment.wordcounter;
-
-import org.example.bon.jova.quarkus.extension.deployment.lyrics.LyricsSanitizer;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
 import java.util.HashMap;
 import java.util.List;

--- a/bon-jova-quarkus-extension/lyrics/src/test/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsReaderTest.java
+++ b/bon-jova-quarkus-extension/lyrics/src/test/java/org/example/bon/jova/quarkus/extension/lyrics/LyricsReaderTest.java
@@ -1,5 +1,6 @@
-package org.example.bon.jova.quarkus.extension.deployment.lyrics;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 import java.util.List;
@@ -11,14 +12,14 @@ class LyricsReaderTest {
     @Test
     void readLyricsShouldReturn100Songs() {
         List<String> allLyrics = LyricsReader.readAll(true);
-        assertEquals(100, allLyrics.size());
+        Assertions.assertEquals(100, allLyrics.size());
     }
 
     @Test
     void readLyricsTwiceShouldUseTheCache() {
         List<String> allLyrics = LyricsReader.readAll(true);
         List<String> allLyrics2 = LyricsReader.readAll(true);
-        assertEquals(100, allLyrics.size());
-        assertEquals(100, allLyrics2.size());
+        Assertions.assertEquals(100, allLyrics.size());
+        Assertions.assertEquals(100, allLyrics2.size());
     }
 }

--- a/bon-jova-quarkus-extension/lyrics/src/test/java/org/example/bon/jova/quarkus/extension/lyrics/WordCounterTest.java
+++ b/bon-jova-quarkus-extension/lyrics/src/test/java/org/example/bon/jova/quarkus/extension/lyrics/WordCounterTest.java
@@ -1,7 +1,6 @@
-package org.example.bon.jova.quarkus.extension.deployment.wordcounter;
+package org.example.bon.jova.quarkus.extension.lyrics;
 
-import org.example.bon.jova.quarkus.extension.deployment.lyrics.LyricsReader;
-import org.example.bon.jova.quarkus.extension.deployment.lyrics.Song;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
@@ -11,7 +10,6 @@ import java.util.Optional;
 import java.util.TreeMap;
 
 import static java.util.Map.entry;
-import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class WordCounterTest {
     @Disabled("See #90")
@@ -23,7 +21,7 @@ class WordCounterTest {
                 .filter(Optional::isPresent)
                 .map(Optional::get)
                 .toList(), true);
-        assertEquals(new TreeMap(expectedWordCounts()), new TreeMap(actualWordCounts));
+        Assertions.assertEquals(new TreeMap(expectedWordCounts()), new TreeMap(actualWordCounts));
     }
 
     private Song sweetChildOfMine() {

--- a/bon-jova-quarkus-extension/pom.xml
+++ b/bon-jova-quarkus-extension/pom.xml
@@ -17,8 +17,9 @@
 
     <modules>
         <module>deployment</module>
-        <module>runtime</module>
         <module>integration-tests</module>
+        <module>runtime</module>
+        <module>lyrics</module>
     </modules>
 
     <properties>

